src/kbc_modules/cc_kbc/attester/tdx/mod.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::Attester;
+use anyhow::*;
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+use tdx_attest_rs;
+
+const CCEL_PATH: &str = "/sys/firmware/acpi/tables/data/CCEL";
+
+pub fn detect_platform() -> bool {
+    Path::new("/dev/tdx-attest").exists()
+}
+
+#[derive(Serialize, Deserialize)]
+struct TdxEvidence {
+    // Base64 encoded CC Eventlog ACPI table
+    // refer to https://uefi.org/specs/ACPI/6.5/05_ACPI_Software_Programming_Model.html#cc-event-log-acpi-table.
+    cc_eventlog: Option<String>,
```

Comment:
Why is this Option? I see that if there's an error reading the CCEL file the whole `get_evidence()` function will return with error (which sounds like the good approach to me). So it seems that there should always be a value for `cc_eventlog`.